share/functions/fish_config.fish

```diff
@@ -205,17 +205,33 @@ function fish_config --description "Launch fish's web based configuration"
                         echo "Too many arguments" >&2
                         return 1
                     end
-                    if not set -q argv[1]
+                    if not set -q argv[1] && contains -- $cmd choose
                         echo "Too few arguments" >&2
                         return 1
                     end
 
+                    # If we are choosing a theme, save its name to a hidden global variable.
+                    # If we are persisting a theme, either get its name from $argv[1] or
+                    # from the last chosen theme.
+                    if contains -- $cmd choose
+                        set -g __fish_last_chosen_theme $argv[1]
+                    else if not set -q argv[1]
+                        if string match -qr . -- $__fish_last_chosen_theme
+                            set argv[1] $__fish_last_chosen_theme
+                            set -ge __fish_last_chosen_theme
+                        else
+                            echo "`fish_config theme save` called before `fish_config theme choose <theme>`"
+                            return 1
+                        end
+                    end
+
                     set -l files $dir/$argv[1].theme
                     set -l file
 
                     set -l scope -g
+
                     if contains -- $cmd save
-                        read -P"Overwrite theme? [y/N]" -l yesno
+                        read -P"Permanently overwrite default theme? [y/N] " -l yesno
```

Comment:
You wouldn't just overwrite the *default* theme. You'd overwrite whatever theme you currently have saved. So I'd just keep the wording here.

Reply:
Okay, then "Overwrite your theme?" or "Overwrite currently saved theme?"?

(also that should be consistent with the above "Overwrite prompt?")

Reply:
That works well.